runtime/commonMain/src/kotlinx/serialization/internal/Primitives.kt

```diff
@@ -70,7 +70,7 @@ private fun checkName(serialName: String) {
 
 @Suppress("UNCHECKED_CAST")
 internal fun <T : Any> KClass<T>.builtinSerializerOrNull(): KSerializer<T>? =
-    BUILTIN_SERIALIZERS[this] as KSerializer<T>
+    BUILTIN_SERIALIZERS[this] as? KSerializer<T>
```

Comment:
It will fail with `ClassCastException: Illegal cast` on JS without any meaningful message, for example.
`builtinSerializerOrNull` is expected to return `null` if there is not serializer and caller will throw more specific exception